### ISSUE TITLE
request

### ISSUE THREAD

**Issue by guyulmaz** [b3ntly/vue-grpc#1]
Hello,
thanks for sharing this project.
is it possible you to make a blog or vlog explaining the stages from start to finish.
for example with which tool how you converted proto file to js, with which you converted from js to truescript.
what stages are needed to make it work with vue js.
at vue.js how easy or right to use this truescript.
at go backend what specific needs exist. because of vue.js?

can you extend this sample with:
1- use vuetify, quasar, onsen frameworks
2- use oauth2 for authentication, authorization with go backend

**Comment by beeekind** [comment]
Definitely, thanks for the interest. This example web application is indeed built with quite a few separate pieces as you mention, some of them are: 

* VueJS: A lightweight single page application framework

* Typescript: A compile-to-javascript language with strong static typing

* Vue-Typed: A small component library that lets you use VueJS in Typescript rather then Javascript 

* Webpack: A build tool that can be used to convert Typescript to Javascript so it can be used in by the browser

* Golang: A simple programming language that is very good for building server applications 

* gRPC: A well maintained and built RPC framework that streamlines client-server communication

* gRPC-web: A tool that allows gRPC to be used from a browser via HTTP/1.1 (gRPC uses HTTP/2)

The quickest answer to your question would be to explore each of these independently. There are good tutorials for all of these out there and once you understand how each part works it becomes easy to wire them together. 

It would be nice to write a tutorial for it at some point but I just don't have the time right now.
